skeleton/backend/src/Backend.hs

```diff
@@ -1,10 +1,13 @@
 module Backend where
 
 import Common.Route
+import Common.Route.Checked
+import Data.Functor.Identity
 import Obelisk.Backend
+import Obelisk.Route
 
 backend :: Backend BackendRoute FrontendRoute
 backend = Backend
   { _backend_run = \serve -> serve $ const $ return ()
-  , _backend_routeEncoder = fullRouteEncoder
+  , _backend_routeEncoder = hoistCheck (pure . runIdentity) validFullEncoder
```

Comment:
This could just be `_backend_routeEncoder = validFullEncoder` but that's a breaking change, probably best to get adoption first (assuming we merge this)